internal/storetesting/charm.go

```diff
@@ -305,6 +305,27 @@ func MetaWithResources(m *charm.Meta, resources ...string) *charm.Meta {
 	return m
 }
 
+// MetaWithDockerResources returns m with Resources set to a set of
+// docker resources with the given names. If m is nil, new(charm.Meta)
+// will be used instead.
+//
+// The description of the resources are derived from the resource name by
+// adding " description" suffix respectively.
+func MetaWithDockerResources(m *charm.Meta, resources ...string) *charm.Meta {
+	if m == nil {
+		m = new(charm.Meta)
+	}
+	m.Resources = make(map[string]resource.Meta)
```

Comment:
I suspect that now we've got two ways of adding resources to metadata that it might be worth respecting the initial Resources value if it's already set (and the same in MetaWithResources).